lib/build-plugin.js

```diff
@@ -32,6 +34,10 @@ function buildPlugin(Client, pluginOpts) {
     const client = new Client(opts.clientOptions)
     const concurrency = opts.concurrency || DEFAULT_GET_CONCURRENCY
 
+    // Register client wrapper
+    fastify.decorate('secretClient', new ClientWrapper(Client, fastify, opts.clientOptions))
```

Comment:
ok let's review this design decision. I would not encapsulate all the existing logic inside the new ClientWrapper class. let's just add a decorator which is a function to refresh secrets (or a specific secret) using the existing client. we may not need to create a new class in the first place